File: go_chunk_app/structversion.go
package main

import (
	"strings"
)

type Chunker struct {
	Sentence string
}

func NewChunker(sentence string) *Chunker {
	return &Chunker{Sentence: sentence}
}

func (c *Chunker) ThreeWordsChunk() string {
	words := strings.Fields(c.Sentence)
	var chunks []string

	for i := 0; i <= len(words)-3; i++ {
		chunks = append(chunks, strings.Join(words[i:i+3], " "))
	}

	return strings.Join(chunks, " | ")
}

File: rust_chunk_app/src/struct_version.rs
// This file contains the struct-based version of the chunking logic

pub struct Chunker {
    sentence: String,
}

impl Chunker {
    pub fn new(sentence: &str) -> Chunker {
        Chunker {
            sentence: sentence.to_string(),
        }
    }

    pub fn three_words_chunk(&self) -> String {
        let words: Vec<&str> = self.sentence.split_whitespace().collect();
        let chunks: Vec<String> = words.windows(3)
            .map(|chunk| chunk.join(" "))
            .collect();
        chunks.join(" | ")
    }
}